### PR TITLE
Uniform null pointer check.

### PR DESCRIPTION
**Pull request by carryxyh**
Uniform null pointer check.

Modification:
Use `ObjectUtil` uniform null pointer check.

Result:
Less code.

### CODE REVIEW

**Comment by carryxyh** [comment]
A big pull request. I reviewed it myself and modified some incorrect checks.

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java [hunk 1]
```diff
@@ -62,9 +63,7 @@ public String getFilename() {
 
     @Override
     public void setFilename(String filename) {
-        if (filename == null) {
-            throw new NullPointerException("filename");
-        }
+        ObjectUtil.checkNotNull(filename, "filename");
         this.filename = filename;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java]: consider merging both lines

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java [hunk 2]
```diff
@@ -93,9 +92,7 @@ public int compareTo(FileUpload o) {
 
     @Override
     public void setContentType(String contentType) {
-        if (contentType == null) {
-            throw new NullPointerException("contentType");
-        }
+        ObjectUtil.checkNotNull(contentType, "contentType");
         this.contentType = contentType;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java]: consider merging both lines

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/MemoryFileUpload.java [hunk 3]
```diff
@@ -56,9 +57,7 @@ public String getFilename() {
 
     @Override
     public void setFilename(String filename) {
-        if (filename == null) {
-            throw new NullPointerException("filename");
-        }
+        ObjectUtil.checkNotNull(filename, "filename");
         this.filename = filename;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/MemoryFileUpload.java]: consider merging both lines

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/MemoryFileUpload.java [hunk 4]
```diff
@@ -87,9 +86,7 @@ public int compareTo(FileUpload o) {
 
     @Override
     public void setContentType(String contentType) {
-        if (contentType == null) {
-            throw new NullPointerException("contentType");
-        }
+        ObjectUtil.checkNotNull(contentType, "contentType");
         this.contentType = contentType;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/MemoryFileUpload.java]: consider merging both lines

**Comment by carryxyh** [comment]
hi @normanmaurer , I checked all checkNotNull calls and merged them. Could you please review it again?

File: buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java [hunk 5]
```diff
@@ -61,14 +62,12 @@
...
-        if (alloc == null) {
-            throw new NullPointerException("alloc");
-        }
         if (maxNumComponents < 1) {
             throw new IllegalArgumentException(
                     "maxNumComponents: " + maxNumComponents + " (expected: >= 1)");
         }
-        this.alloc = alloc;
+
+        this.alloc = ObjectUtil.checkNotNull(alloc, "alloc");
```
Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java]: @carryxyh lets move this check (and assignment) above the `maxNumComponents` check again (just like before).

File: codec-socks/src/main/java/io/netty/handler/codec/socksx/v5/DefaultSocks5CommandRequest.java [hunk 6]
```diff
@@ -63,7 +57,7 @@ public DefaultSocks5CommandRequest(
             throw new IllegalArgumentException("dstPort: " + dstPort + " (expected: 0~65535)");
         }
 
-        this.type = type;
+        this.type = ObjectUtil.checkNotNull(type, "type");
```
Comment (normanmaurer) [inline codec-socks/src/main/java/io/netty/handler/codec/socksx/v5/DefaultSocks5CommandRequest.java]: lets move the check and assignment before the other stuff to keep the behaviour the same as before.

File: codec/src/main/java/io/netty/handler/codec/DecoderResult.java [hunk 7]
```diff
@@ -26,19 +27,14 @@
     public static final DecoderResult SUCCESS = new DecoderResult(SIGNAL_SUCCESS);
 
     public static DecoderResult failure(Throwable cause) {
-        if (cause == null) {
-            throw new NullPointerException("cause");
-        }
+        ObjectUtil.checkNotNull(cause, "cause");
         return new DecoderResult(cause);
```
Comment (normanmaurer) [inline codec/src/main/java/io/netty/handler/codec/DecoderResult.java]: nit: merge lines

File: codec/src/main/java/io/netty/handler/codec/LengthFieldBasedFrameDecoder.java [hunk 8]
```diff
@@ -319,7 +316,7 @@ public LengthFieldBasedFrameDecoder(
                     "lengthFieldLength (" + lengthFieldLength + ").");
         }
 
-        this.byteOrder = byteOrder;
+        this.byteOrder = checkNotNull(byteOrder, "byteOrder");
```
Comment (normanmaurer) [inline codec/src/main/java/io/netty/handler/codec/LengthFieldBasedFrameDecoder.java]: move the check and assignment before the other checks to ensure the behaviour is not changed.

File: common/src/main/java/io/netty/util/ThreadDeathWatcher.java [hunk 9]
```diff
@@ -77,31 +78,22 @@
...
-            throw new NullPointerException("task");
-        }
+        ObjectUtil.checkNotNull(thread, "thread");
+
         if (!thread.isAlive()) {
             throw new IllegalArgumentException("thread must be alive.");
         }
 
-        schedule(thread, task, true);
+        schedule(thread, ObjectUtil.checkNotNull(task, "task"), true);
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/ThreadDeathWatcher.java]: move the mull check under the null check for thread to ensure we not change behaviour.

File: common/src/main/java/io/netty/util/ThreadDeathWatcher.java [hunk 10]
```diff
@@ -137,13 +129,10 @@ public Void run() {
...
     public static boolean awaitInactivity(long timeout, TimeUnit unit) throws InterruptedException {
-        if (unit == null) {
-            throw new NullPointerException("unit");
-        }
-
         Thread watcherThread = ThreadDeathWatcher.watcherThread;
         if (watcherThread != null) {
-            watcherThread.join(unit.toMillis(timeout));
+            watcherThread.join(
+                    ObjectUtil.checkNotNull(unit, "unit").toMillis(timeout));
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/ThreadDeathWatcher.java]: move the null check as first action in the method to ensure we also throw if `watcherThread == null` and `unit` is null

File: common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java [hunk 11]
```diff
@@ -167,15 +167,16 @@ protected final boolean hasScheduledTasks() {
...
         ObjectUtil.checkNotNull(unit, "unit");
         if (delay < 0) {
             delay = 0;
         }
         validateScheduled0(delay, unit);
 
         return schedule(new ScheduledFutureTask<Void>(
-                this, command, deadlineNanos(unit.toNanos(delay))));
+                this,
+                ObjectUtil.checkNotNull(command, "command"),
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java]: move null check back as first action to not change behaviour.

File: common/src/main/java/io/netty/util/concurrent/GlobalEventExecutor.java [hunk 12]
```diff
@@ -192,25 +190,17 @@ public boolean awaitTermination(long timeout, TimeUnit unit) {
...
-        if (unit == null) {
-            throw new NullPointerException("unit");
-        }
-
         final Thread thread = this.thread;
         if (thread == null) {
             throw new IllegalStateException("thread was not started");
         }
-        thread.join(unit.toMillis(timeout));
+        thread.join(ObjectUtil.checkNotNull(unit, "unit").toMillis(timeout));
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/concurrent/GlobalEventExecutor.java]: move null check back as first action in the method to not change behaviour

File: common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java [hunk 13]
```diff
@@ -811,15 +802,11 @@ protected boolean confirmShutdown() {
...
-        if (unit == null) {
-            throw new NullPointerException("unit");
-        }
-
         if (inEventLoop()) {
             throw new IllegalStateException("cannot await termination of the current thread");
         }
 
-        threadLock.await(timeout, unit);
+        threadLock.await(timeout, ObjectUtil.checkNotNull(unit, "unit"));
```
Comment (normanmaurer) [inline common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java]: move check back as first in the method to not change behaviour if null is passed as `unit` but we are in the `EventLoop` thread.

File: example/src/main/java/io/netty/example/worldclock/WorldClockProtocol.java [hunk 14]
```diff
@@ -19,6 +19,8 @@
 
 package io.netty.example.worldclock;
 
+import io.netty.util.internal.ObjectUtil;
```
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/worldclock/WorldClockProtocol.java]: please not change this class as it was generated.

File: handler/src/main/java/io/netty/handler/stream/ChunkedFile.java [hunk 15]
```diff
@@ -82,27 +83,13 @@ public ChunkedFile(RandomAccessFile file, int chunkSize) throws IOException {
...
-        if (chunkSize <= 0) {
-            throw new IllegalArgumentException(
-                    "chunkSize: " + chunkSize +
-                    " (expected: a positive integer)");
-        }
+        ObjectUtil.checkPositiveOrZero(offset, "offset");
+        ObjectUtil.checkPositiveOrZero(length, "length");
 
-        this.file = file;
+        this.file = ObjectUtil.checkNotNull(file, "file");
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/stream/ChunkedFile.java]: move this assignment and check as first in the method to not change behaviour.

File: resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java [hunk 16]
```diff
@@ -115,8 +115,9 @@ public void clear() {
 
     @Override
     public boolean clear(String hostname) {
-        checkNotNull(hostname, "hostname");
-        return resolveCache.clear(appendDot(hostname));
+        return resolveCache.clear(
+                appendDot(
+                        checkNotNull(hostname, "hostname")));
```
Comment (normanmaurer) [inline resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java]: nit: maybe merge the lines as the formatting looks a bit strange.

File: transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDatagramChannel.java [hunk 17]
```diff
@@ -260,7 +245,7 @@ protected void doBind(SocketAddress localAddress) throws Exception {
 
     @Override
     protected void doWrite(ChannelOutboundBuffer in) throws Exception {
-        for (;;) {
+        for (; ; ) {
```
Comment (normanmaurer) [inline transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDatagramChannel.java]: nit: please revert

File: transport/src/main/java/io/netty/channel/DefaultAddressedEnvelope.java [hunk 18]
```diff
@@ -39,15 +40,11 @@
...
-        if (message == null) {
-            throw new NullPointerException("message");
-        }
-
         if (recipient == null && sender == null) {
             throw new NullPointerException("recipient and sender");
         }
 
-        this.message = message;
+        this.message = ObjectUtil.checkNotNull(message, "message");
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/DefaultAddressedEnvelope.java]: move this assignment and check as first in the method to not change behaviour

File: transport/src/main/java/io/netty/channel/DefaultFileRegion.java [hunk 19]
```diff
@@ -71,14 +67,9 @@ public DefaultFileRegion(FileChannel file, long position, long count) {
...
     public DefaultFileRegion(File f, long position, long count) {
-        if (f == null) {
-            throw new NullPointerException("f");
-        }
-        checkPositiveOrZero(position, "position");
-        checkPositiveOrZero(count, "count");
-        this.position = position;
-        this.count = count;
-        this.f = f;
+        this.f = ObjectUtil.checkNotNull(f, "file");
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/DefaultFileRegion.java]: nit: this should be "f"

File: transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java [hunk 20]
```diff
@@ -91,14 +91,10 @@ public ChannelFuture register(final ChannelPromise promise) {
...
-        }
-        if (promise == null) {
-            throw new NullPointerException("promise");
-        }
-
-        channel.unsafe().register(this, promise);
+        ObjectUtil.checkNotNull(promise, "promise");
+        ObjectUtil.checkNotNull(channel, "channel").
+                unsafe().
+                register(this, promise);
```
Comment (normanmaurer) [inline transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java]: nit consider merging some lines... The formatting looks strange.

Comment (carryxyh) [reply transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java]: use this version. this looks better..

```java
ObjectUtil.checkNotNull(promise, "promise");
ObjectUtil.checkNotNull(channel, "channel");
channel.unsafe().register(this, promise);
```

File: resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java (hunk 16 above)
Comment (carryxyh) [reply resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java]: revert, use this version:
```
checkNotNull(hostname, "hostname");
return resolveCache.clear(appendDot(hostname));
```

**Comment by carryxyh** [comment]
@normanmaurer  fixed. pls check.

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@carryxyh please fix test errors.

**Comment by carryxyh** [comment]
@netty-bot test this please

**Comment by carryxyh** [comment]
@normanmaurer Fixed.
It seems netty-bot doesn't follow my instructions. :)

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@carryxyh thanks a lot!

**Pull request merged**